vnext/Shared/WinRTWebSocketResource.cpp

```diff
@@ -330,6 +298,35 @@ void WinRTWebSocketResource::Synchronize() noexcept {
 #pragma region IWebSocketResource
 
 void WinRTWebSocketResource::Connect(string &&url, const Protocols &protocols, const Options &options) noexcept {
+  m_socket.MessageReceived(
```

Comment:
Note, the event is set here instead of the constructor to be able to call `shared_from_this`.

Reply:
Is there a chance that there would be multiple connections on this object?
I didn't see  a 'disconnect' or 'reconnect' api. But there is a close.

What if caller users connect, close, connect?
I don't see any guard against that state int he connect function...

Reply:
Not quite.

Based on the WebSockets spec, they can only be connected and closed once.

If a connection is lost/broken, a new WebSocket should be spun.

Reply:
We could add the guard, but this is usually managed at upper levels (i.e. CxxModule or JavaScript), so I think it could be overkill.